pkg/raft/raft.go

```diff
@@ -2473,6 +2473,11 @@ func (r *raft) handleSnapshot(m pb.Message) {
 	if r.restore(s) {
 		r.logger.Infof("%x [commit: %d] restored snapshot [index: %d, term: %d]",
 			r.id, r.raftLog.committed, id.index, id.term)
+		// This can happen if we receive a snapshot from leader of previous term.
```

Comment:
Need to elaborate this comment a bit. Specifically, why it's safe to send this message to another leader.

Reply:
The safety precondition for sending `MsgAppResp` is being sure that our log is consistent with the leader's log.

Since we're sending the reply to a leader `r.lead` with a greater term than the snapshot sender's, the `r.lead` necessarily contains all the committed entries in its log. This makes `MsgAppResp` correct w.r.t. to this new leader.

From raft paper:

```
Leader Completeness: if a log entry is committed in a
given term, then that entry will be present in the logs
of the leaders for all higher-numbered terms. §5.4
```